telephony/android_modem.c

```diff
@@ -1935,24 +1977,8 @@ handleRadioPower( const char*  cmd, AModem  modem )
         return "+CME ERROR: 50";
     }
 
-    if (radio_state == modem->radio_state) {
-        return "OK";
-    }
-
-    modem->radio_state = radio_state;
     amodem_reply(modem, "OK");
-
-    switch (radio_state) {
-        case A_RADIO_STATE_OFF:
-            amodem_set_voice_registration(modem, A_REGISTRATION_UNREGISTERED);
-            amodem_set_data_registration(modem, A_REGISTRATION_UNREGISTERED);
-            asimcard_reset_status_after_radio_off(modem->sim);
-            break;
-        case A_RADIO_STATE_ON:
-            amodem_set_voice_registration(modem, A_REGISTRATION_HOME);
-            amodem_set_data_registration(modem, A_REGISTRATION_HOME);
-            break;
-    }
+    _amodem_set_radio_state(modem, radio_state);
```

Comment:
With this changes, __RIL_UNSOL_RESPONSE_RADIO_STATE_CHANGED__ will be sent twice, one is sent by [reference_ril](https://github.com/mozilla-b2g/platform_hardware_ril/blob/master/reference-ril/reference-ril.c#L3377-L3419) itself after success to switch radio state.